ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/models/ComplexType.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2015 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen.c.models;
+
+/**
+ * A ComplexType indicates a StructMember that requires a free,
+ * in other words a Struct inside of a Struct.
+ */
+public class ComplexType extends C_Type {
```

Comment:
Instead of calling this ComplexType, I would call it something like, CustomStructType, or something else.  I don't think Complex is really conveying the right message about the type.  Maybe FreeableType would be better than CustomStructType.